sherpa/optmethods/optfcts.py

```diff
@@ -228,6 +229,118 @@ def _set_limits(x, xmin, xmax):
     return 0
 
 
+class NoTransformation:
+    def __init__(self):
+        return
+
+    def calc_covar(self, m, n, xint, info, fjac):
+        covar = None
+        if info > 0:
+            fjac = numpy.reshape(numpy.ravel(fjac, order='F'),
+                                 (m, n), order='F')
+            if m != n:
+                covar = fjac[:n, :n]
+            else:
+                covar = fjac
+        return covar
+
+    def chain_rule(self, x):
+        return numpy.ones_like(x)
+
+    def ext2int(self, x):
+        return x
+
+    def int2ext(self, x):
+        return x
+
+
+class Transformation(NoTransformation):
+
+    def __init__(self, lo, hi):
+        self.lo = lo
+        self.hi = hi
+        self.bounds, self.int2ext_fcns, self.ext2int_fcns, = self.init()
+        return
+
+    def init(self):
+        bounds = []
```

Comment:
Writing down what @dtnguyen2 says in words right now, but I can't type as fast as you speak, so not sure it's all right.
```suggestion
    def init(self):
        '''Decide which demonstrations to do.
        
        This code selects which transformations to do. This is done per parameter, since 
        some parameters might be bounded, while others are unbounded.
        There are four different scenarios:
        
        - A parameter is unbounded.
        - A parameter has an upper bound.
        - A parameter has a lower bound.
        - A parameter has both a lower and and upper bound.
        
        Transformations are only needed if a parameter is close to the boundaries and 
        there is a performance penalty to using it.
        If parameters are close to the max value stored in a float, any transformation will cause a loss of precision.
        '''
        bounds = []
```